### PR TITLE
RavenDB-21027 Reset bitset for each document in `AllIn`.

### PR DESCRIPTION
**Pull request by maciejaszyk**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21027 
### Type of change

- Bug fix

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Corax/Queries/UnaryMatch.Ordinal.cs [hunk 1]
```diff
@@ -137,13 +112,15 @@ private static int BinarySearch(TermQueryItem[] item, ReadOnlySpan<byte> value)
             return -1;
         }
 
-        private static unsafe int FillFuncSequenceAllIn(ref UnaryMatch<TInner, TValueType> match, Span<long> matches)
+        [SkipLocalsInit]
```
Comment (ayende) [inline src/Corax/Queries/UnaryMatch.Ordinal.cs:115]: Remove this

File: src/Corax/Queries/UnaryMatch.Ordinal.cs [hunk 2]
```diff
@@ -137,13 +112,15 @@ private static int BinarySearch(TermQueryItem[] item, ReadOnlySpan<byte> value)
...
+        private static int FillFuncSequenceAllIn(ref UnaryMatch<TInner, TValueType> match, Span<long> matches)
         {
             var value = ((TermQueryItem[])(object)match._value);
             var requiredSizeOfBitset = value.Length / sizeof(byte) + (value.Length % sizeof(byte) == 0 ? 0 : 1);
-            byte* bitsetBuffer = stackalloc byte[requiredSizeOfBitset];
-            new Span<byte>(bitsetBuffer, requiredSizeOfBitset).Clear();
             
+            //We're skipping init here since we will do it below (only when Inner match has values).
+            byte* bitsetBuffer = stackalloc byte[requiredSizeOfBitset];
```
Comment (ayende) [inline src/Corax/Queries/UnaryMatch.Ordinal.cs:122]: Remove this

Comment (redknightlois) [reply src/Corax/Queries/UnaryMatch.Ordinal.cs:122]: Why? We are clearing it on every iteration.

Comment (redknightlois) [reply src/Corax/Queries/UnaryMatch.Ordinal.cs:122]: For context: https://github.com/ravendb/ravendb/commit/be9759537e7ea576df518c5eb6f47221298c8223#commitcomment-124651334

There is a big misconception regarding guarantees about initialization (I explained to @maciejaszyk why the documentation is just plain wrong when dealing with C#). C# code is guaranteed to initialize the stackalloc to `0` UNLESS you specify that you are going to be doing it yourself. Incorrect removal of the `locals .init` marker on methods is treated as a compiler bug. https://github.com/dotnet/runtime/issues/43391#issuecomment-708473741

EDIT: That's why I asked Maciej to comment it to make it unambiguous why we were doing it that way.

Comment (ayende) [reply src/Corax/Queries/UnaryMatch.Ordinal.cs:122]: Okay

**Pull request merged**